packages/contracts/contracts/registryManager/RegistryManager.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+import { Common } from "../common/Common.sol";
+import { IRecipientRegistry } from "../interfaces/IRecipientRegistry.sol";
+import { IRegistryManager } from "../interfaces/IRegistryManager.sol";
+
+/// @title RegistryManager
+/// @notice Contract that allows to use send, approve, reject requests to RecipientRegistry.
+contract RegistryManager is Ownable, IRegistryManager, Common {
+  /// @notice requests
+  mapping(uint256 => Request) internal requests;
+
+  /// @notice request statuses
+  mapping(uint256 => Status) internal statuses;
+
+  /// @inheritdoc IRegistryManager
+  uint256 public requestCount;
+
+  /// @notice Initialize registry manager
+  constructor() payable Ownable(msg.sender) {}
+
+  /// @notice Check if request is valid
+  modifier isValidRequest(Request calldata request) {
+    if (request.registry == address(0)) {
+      revert ValidationError();
+    }
+
+    if (request.recipient.recipient == address(0)) {
+      revert ValidationError();
+    }
+
+    uint256 count = IRecipientRegistry(request.registry).recipientCount();
+
+    if (request.index >= count && request.requestType == RequestType.Change) {
+      revert ValidationError();
+    }
+
+    _;
+  }
+
+  /// @notice Check if request is pending and exists
+  /// @param index Request index
+  modifier isPending(uint256 index) {
+    if (index >= requestCount || statuses[index] != Status.Pending) {
+      revert OperationError();
+    }
+    _;
+  }
+
+  /// @inheritdoc IRegistryManager
+  function send(Request calldata request) public virtual override isValidRequest(request) {
+    requests[requestCount] = request;
```

Comment:
do you think we can condense this into one mapping only? we could add the status directly to the request, which by default when created would be Pending

Reply:
In this case, it should be a different struct because user can send different options for status and it's not good to validate this case on contracts side.

Reply:
no need to validate, you could forcefully set it to Pending (change calldata to memory) - one extra operation there, but we get rid of a mapping and all extra write and read